enterprise/cmd/frontend/internal/auth/gitlaboauth/session.go

```diff
@@ -56,6 +57,13 @@ func (s *sessionIssuerHelper) GetOrCreateUser(ctx context.Context, token *oauth2
 		return nil, message, errors.New(message)
 	}
 
+	// AllowSignup defaults to true, unlinke with GitHub. If not set, we will create an account for the user with their GitLab email address.
+	// If false, a new account won't be created for the GitLab user.
```

Comment:
nit: I don't think we need to explain more than we do here
```suggestion
	// AllowSignup defaults to true when not set to preserve the existing behavior.
```